src/firmware/app/primitives/shoot_alignment.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "firmware/app/world/firmware_world.h"
+
+/**
+ * Start the shoot alignment in the given world
+ * @param void_state_ptr [in] A pointer to this primitives state, as allocated by the
+ *                            `create_state` function
+ * @param world [in] The world this primitive is running in.
+ * @param x_destination The x destination for shooting
+ * @param y_destination The y destination for shooting
+ * @param alignment_angle The angle to align to
+ */
+void app_shoot_alignment_start(void *void_state_ptr, FirmwareWorld_t *world,
```

Comment:
Never thought about calling a primitive from another primitive :brain: 

May I suggest `chick_aligment` as a more generic name since it's used by both kick and chip? Also this primitive actually does technically do the kicking, so not sure if the name should reflect that? If not a comment is probably sufficient since this just handles the motion and kick/chip handle the actual chick actuation.

Reply:
I can change the name. If kick_primitive or chip_primitive doesn't set up autokick/autochip then it won't kick/chip, so it's entirely dependent on that

Reply:
Yea my point is more that "alignment" would make me think the robot lines up the ball but doesn't move forward to make contact with it. This isn't super blocking but would like thoughts from other people as to how intuitive this is on first-read. @kelvinkoon @jared-paul @LiCody @HaomiaoZ 

Other possible names?
* `app_chick_movement_start`
* `app_chick_motion_start`

Reply:
IMO `app_chick_motion_start` is the most intuitive.

Reply:
`app_chick_move_start` makes to most sense to me

Reply:
I'd vote for `app_chick_motion_start` and making sure that we have an ascii diagram at the start of this file explaining what it does.

Reply:
`app_chick_motion_start` wins with 2 votes